cubed/array_api/array_object.py

```diff
@@ -31,6 +37,55 @@ def __array__(self, dtype=None) -> np.ndarray:
     def __repr__(self):
         return f"cubed.Array<{self.name}, shape={self.shape}, dtype={self.dtype}, chunks={self.chunks}>"
 
+    def _repr_html_(self):
+        try:
+            grid = self.to_svg(size=ARRAY_SVG_SIZE)
+        except NotImplementedError:
+            grid = ""
+
+        if not math.isnan(self.nbytes):
+            nbytes = format_bytes(self.nbytes)
+            cbytes = format_bytes(math.prod(self.chunksize) * self.dtype.itemsize)
+        else:
+            nbytes = "unknown"
+            cbytes = "unknown"
+
+        return get_template("array.html.j2").render(
+            array=self,
+            grid=grid,
+            nbytes=nbytes,
+            cbytes=cbytes,
+            arrs_in_plan=f"{self.plan.dag.number_of_nodes()} arrays in Plan",
```

Comment:
Wasn't quite sure if this was the correct way to get the number of arrays in the plan.

Reply:
This will give the number of arrays in the unoptimized plan. Maybe add a `num_arrays(optimize_graph=True)` method to `Plan` (to mirror the `num_tasks` method)?